content/docs/1.4.0/advanced-resources/rwx-workloads.md

```diff
@@ -1,60 +1,69 @@
+title: ReadWriteMany (RWX) Volume
 weight: 4
 ---
 
-Longhorn natively supports RWX workloads, by exposing a regular Longhorn volume via a NFSv4 server (share-manager).
+Longhorn supports ReadWriteMany (RWX) volumes by exposing regular Longhorn volumes via NFSv4 servers that reside in share-manager pods.
 
-The following diagram shows how the RWX support works:
 
-{{< figure src="/img/diagrams/rwx/rwx-native-architecture.png" >}}
+# Introduction
 
-For each actively in use RWX volume Longhorn will create a `share-manager-<volume-name>` Pod in the `longhorn-system` namespace.
+For each actively in use RWX volume Longhorn will create a `share-manager-<volume-name>` Pod in the `longhorn-system` namespace. This Pod is responsible for exporting a Longhorn volume via a NFSv4 server that is running inside the Pod. There is also a service created for each RWX volume, and that is used as an endpoint for the actual NFSv4 client connection.
 
-This Pod is responsible for exporting a Longhorn volume via a NFSv4 server that is running inside the Pod.
-
-There is also a service created for each RWX volume, and that is used as an endpoint for the actual NFSv4 client connection.
+{{< figure src="/img/diagrams/rwx/rwx-arch.png" >}}
 
 # Requirements
 
-To be able to use RWX volumes, each client node needs to have a NFSv4 client installed.
+It is necessary to meet the following requirements in order to use RWX volumes.
 
-For Ubuntu you can install a NFSv4 client via:
+1. Each NFS client node needs to have a NFSv4 client installed.
+     - For Debian based distros you can install a NFSv4 client via:
+        ```
+        apt install nfs-common
+        ```
 
-```
-apt install nfs-common
-```
+     - For RPM based distros you can install a NFSv4 client via:
+        ```
+        yum install nfs-utils
+        ```
 
-For RPM based distros you can install a NFSv4 client via:
+      > **Troubleshooting:** If the NFSv4 client is not available on the node, when trying to mount the volume the below message will be part of the error:
+      > ```
+      > for several filesystems (e.g. nfs, cifs) you might need a /sbin/mount.<type> helper program.
+      > ```
 
-```
-yum install nfs-utils
-```
+2. The hostname of each node is unique in the Kubernetes cluster.
 
-If the NFSv4 client is not available on the node, when trying to mount the volume the below message will be part of the error:
-```
-for several filesystems (e.g. nfs, cifs) you might need a /sbin/mount.<type> helper program.\n
-```
+    There is a dedicated recovery backend service for NFS servers in Longhorn system. When a client connects to an NFS server, the client's information, including its hostname, will be stored in the recovery backend. When a share-manager Pod or NFS server is abnormally terminated, Longhorn will create a new one. Within the 90-seconds grace period, clients will reclaim locks using the client information stored in the recovery backend.
```

Comment:
```suggestion
    There is a dedicated recovery backend service for the NFS server in each share manager pod in the Longhorn system. When a client connects to an NFS server, the client's information, including its hostname, will be stored in the recovery backend. When a share-manager Pod or the internal NFS server is abnormally terminated, Longhorn will create a new one. Within the 90-seconds grace period, clients will reclaim locks using the client information stored in the recovery backend.
```

Reply:
The hostname reported by the NFS server should be the hostname of the share manager pod which is the name of the share manager CR, so why does the hostname of each Node matter?

I assume the hostname will be the same after the share manager pod failed over because the name should be still the same as the share manager CR. 

Anything missing?

Reply:
Yeah, you are right.

About `So the client id, that the NFS server received from the client, is actually the hostname of the client’s host`, if more than one pods mount the same RWX volume, what the client ID will be for each pod which is on the same node? How to differentiate them? (I assume the client ID is composed of the host address + <id>?)

Reply:
IIRC, because we use NFS4.1, so there is only one `session` between the server node and the client node. All connections between them use the same session. Before creating the session, the client id is exchanged. Thus, all clients on a node use the same client id.

Reply:
I see, then it's clarified :+1:

Reply:
> I assume the client ID is composed of the host address + ?

Yes, a generic NFS client id is `host name + ip of the client host`. But we will encounter an [issue](https://github.com/longhorn/longhorn/issues/2293#issuecomment-1194072597) if the ` ip of the client host` is added, so the `ip of the client host` is removed and only uses the host name.